bl-plugins/api/plugin.php

```diff
@@ -68,6 +68,14 @@ public function beforeAll()
 		// METHOD
 		// ------------------------------------------------------------
 		$method = $this->getMethod();
+		
+		// Added full CORS policy and OPTIONS method to allow API call from JS
+		if ($method === 'OPTIONS'){ 
+		      header('Access-Control-Allow-Origin: *');
+		      header('Access-Control-Allow-Methods: GET, HEAD, OPTIONS, POST, PUT');
+		      header("Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, Authorization");
+		      exit;
```

Comment:
I don't get it, when the method is `OPTIONS` will exit ?

Reply:
Looks like for the OPTIONS should be HTTP 204.

$this->response(204, 'No Content');

Reply:
Yes, now on OPTIONS exit with 404 and wrong CORS policy. 204 is also valid response.

Reply:
I was checking here the 204. https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
Where you get the 404 ?

Reply:
When api is called from javascript browser first send Options request to get permissions, and then send actual get/post/put request. On this first "options" request api return error 400 (sorry) because it is empty https://github.com/bludit/bludit/blob/5abddff1aa16f3e408bc4ee66832305d86a6705c/bl-plugins/api/plugin.php#L74 and no inputs.